src/Language/PureScript/TypeChecker.hs

```diff
@@ -712,3 +713,18 @@ typeCheckModule (Module ss coms mn decls (Just exps)) =
     extractMemberName (TypeDeclaration td) = tydeclIdent td
     extractMemberName _ = internalError "Unexpected declaration in typeclass member list"
   checkClassMembersAreExported _ = return ()
+
+  checkDataConstructorsAreExported :: DeclarationRef -> m ()
```

Comment:
Would you mind adding a comment here? I think something along the lines of "if any data constructors are exported, we require all data constructors to be exported" would be enough.

Reply:
Sure! I added a comment.